snapcraft/_options.py

```diff
@@ -156,6 +156,13 @@ def target_arch(self):
     @property
     def cross_compiler_prefix(self):
         try:
+            # cross-compilation of x86 32bit binaries on a x86_64 host is
+            # possible by reusing the native toolchain - let Kbuild figure
+            # it out by itself and pass down an empy cross-compiler-prefix
+            # to start the build
+            if (self.__platform_arch == 'x86_64' and
+                    self.__target_machine == 'i686'):
+                return
```

Comment:
This doesn't just apply to the kernel though, right? Do the other build systems handle this appropriately? @kalikiana that may be a question for you.

Reply:
It's only used by the kbuild plugin which the kernel plugin subclasses afair. Nothing else uses that value.

Reply:
`self.project.arch_triplet` is used elsewhere - it may be something to consider for the future if it should be re-used in other places, I don't recall if there was a specific reason for this.

Reply:
Alright, +1 then.

Reply:
Either one is good. But as this came up twice, I changed it to ''.